index.js

```diff
@@ -58,8 +58,24 @@ function getMainArgs() {
   return ArrayPrototypeSlice(process.argv, 2);
 }
 
-function storeOptionValue(options, longOption, value, result) {
-  const optionConfig = options[longOption] || {};
+function storeOptionValue(strict, options, longOption, value, result) {
+  const hasOptionConfig = ObjectHasOwn(options, longOption);
+
+  if (strict) {
+    if (!hasOptionConfig) {
+      throw new Error(`Unknown option: --${longOption}`);
+    }
+
+    if (options[longOption].type === 'string' && value == null) {
```

Comment:
I think it would be clearer to test against `undefined` rather than `null`, since the value is `undefined`.

Reply:
https://github.com/pkgjs/parseargs/pull/74/commits/ca05b430e41c3056c367e6169329b5ea42d21d0b

Reply:
`== null` checks for both, which is the correct semantic check imo - just like `??` and optional chaining work.

Reply:
I understand that it covers both, but I still found it odd to test against `null` when that is not a value that we expect whereas `undefined` is. Good comparison with `??` and optional chaining. I didn't say how strongly to compare (and hadn't though that through!), and with that in mind suggest:

```
value == undefined
```

Reply:
Thanks, I had forgotten that possibility. (I haven't hit it in much code so not idiomatic in my mind yet!)

Reply:
> since undefined is an identifier.

Checking my understanding. Perhaps not a practical problem in modern contexts since ECMAScript 5 made undefined read-only? (I was hacking about in node REPL and saw the read-only behaviour, went reading to find out why.)

Reply:
(I used `undefined` in a few other places in the code already outside this PR, which I will update to follow whatever pattern we use here.)

Reply:
ES5 made `undefined` readonly _only in the global scope_. it's fully overrideable everywhere else, including inside a CJS module's scope. The module wrapper can also be altered at runtime via `require.extensions`, so it's a real risk in node.

Reply:
(Thanks for the explanation. Appreciated.)

Reply:
(I will review use of `undefined` identifier outside `storeOptionValue` in #75, won't touch `storeOptionValue`.)

Reply:
https://github.com/pkgjs/parseargs/pull/74#discussion_r846578532